<!DOCTYPE html>
<html lang='en'>
<head>
<title>xvidcap-ffmpeg.patch - aur.git - AUR Package Repositories</title>
<meta name='generator' content='cgit v1.2.3-2-g540e'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/css/cgit.css'/>
<link rel='shortcut icon' href='/images/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://aur.archlinux.org/cgit/aur.git/atom/xvidcap-ffmpeg.patch?h=xvidcap' type='application/atom+xml'/>
<link rel='vcs-git' href='https://aur.archlinux.org/xvidcap.git/' title='aur.git Git repository'/>
</head>
<body>
		<div id="archnavbar" class="anb-aur">
			<div id="archnavbarlogo"><h1><a href="/" title="Return to the main page">Arch Linux User Repository</a></h1></div>
			<div id="archnavbarmenu">
				<ul id="archnavbarlist">
					<li id="anb-home"><a href="https://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
					<li id="anb-packages"><a href="https://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
					<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
					<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
					<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and track bugs">Bugs</a></li>
					<li id="anb-security"><a href="https://security.archlinux.org/" title="Arch Linux Security Tracker">Security</a></li>
					<li id="anb-aur"><a href="/" title="Arch Linux User Repository">AUR</a></li>
					<li id="anb-download"><a href="https://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
				</ul>
			</div>
		</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'>AUR : xvidcap.git</td></tr>
<tr><td class='sub'>AUR Package Repositories | click <a href='/pkgbase/xvidcap/'>here</a> to return to the package base details page</td><td class='sub right'></td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/aur.git/?h=xvidcap'>summary</a><a href='/cgit/aur.git/log/xvidcap-ffmpeg.patch?h=xvidcap'>log</a><a class='active' href='/cgit/aur.git/tree/xvidcap-ffmpeg.patch?h=xvidcap'>tree</a><a href='/cgit/aur.git/commit/xvidcap-ffmpeg.patch?h=xvidcap'>commit</a><a href='/cgit/aur.git/diff/xvidcap-ffmpeg.patch?h=xvidcap'>diff</a><a href='/cgit/aur.git/stats/xvidcap-ffmpeg.patch?h=xvidcap'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/aur.git/log/xvidcap-ffmpeg.patch'>
<input type='hidden' name='h' value='xvidcap'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='search' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/aur.git/tree/?h=xvidcap'>root</a>/<a href='/cgit/aur.git/tree/xvidcap-ffmpeg.patch?h=xvidcap'>xvidcap-ffmpeg.patch</a></div><div class='content'>blob: 3a74bf508bc8afa85df60c163d609cf61650054d (<a href='/cgit/aur.git/plain/xvidcap-ffmpeg.patch?h=xvidcap'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
<a id='n89' href='#n89'>89</a>
<a id='n90' href='#n90'>90</a>
<a id='n91' href='#n91'>91</a>
<a id='n92' href='#n92'>92</a>
<a id='n93' href='#n93'>93</a>
<a id='n94' href='#n94'>94</a>
<a id='n95' href='#n95'>95</a>
<a id='n96' href='#n96'>96</a>
<a id='n97' href='#n97'>97</a>
<a id='n98' href='#n98'>98</a>
<a id='n99' href='#n99'>99</a>
<a id='n100' href='#n100'>100</a>
<a id='n101' href='#n101'>101</a>
<a id='n102' href='#n102'>102</a>
<a id='n103' href='#n103'>103</a>
<a id='n104' href='#n104'>104</a>
<a id='n105' href='#n105'>105</a>
</pre></td>
<td class='lines'><pre><code>Index: src/codecs.c
===================================================================
<span class="hl kwa">--- src/codecs.c	(revision 319)</span>
<span class="hl kwb">+++ src/codecs.c	(working copy)</span>
<span class="hl kwd">&#64;&#64; -43,8 +43,8 &#64;&#64;</span>
 #include &lt;locale.h&gt;
 
 #ifdef USE_FFMPEG
<span class="hl kwa">-#include &lt;ffmpeg/avcodec.h&gt;</span>
<span class="hl kwa">-#include &lt;ffmpeg/avformat.h&gt;</span>
<span class="hl kwb">+#include &lt;libavcodec/avcodec.h&gt;</span>
<span class="hl kwb">+#include &lt;libavformat/avformat.h&gt;</span>
 #endif     // USE_FFMPEG
 
 #include &quot;app_data.h&quot;
Index: src/xtoffmpeg.c
===================================================================
<span class="hl kwa">--- src/xtoffmpeg.c	(revision 319)</span>
<span class="hl kwb">+++ src/xtoffmpeg.c	(working copy)</span>
<span class="hl kwd">&#64;&#64; -56,13 +56,13 &#64;&#64;</span>
 #include &quot;xvidcap-intl.h&quot;
 
 // ffmpeg stuff
<span class="hl kwa">-#include &lt;ffmpeg/avcodec.h&gt;</span>
<span class="hl kwa">-#include &lt;ffmpeg/avformat.h&gt;</span>
<span class="hl kwa">-#include &lt;ffmpeg/avdevice.h&gt;</span>
<span class="hl kwb">+#include &lt;libavcodec/avcodec.h&gt;</span>
<span class="hl kwb">+#include &lt;libavformat/avformat.h&gt;</span>
<span class="hl kwb">+#include &lt;libavdevice/avdevice.h&gt;</span>
 //#include &lt;ffmpeg/dsputil.h&gt;
<span class="hl kwa">-#include &lt;ffmpeg/swscale.h&gt;</span>
<span class="hl kwa">-#include &lt;ffmpeg/rgb2rgb.h&gt;</span>
<span class="hl kwa">-#include &lt;ffmpeg/fifo.h&gt;</span>
<span class="hl kwb">+#include &lt;libswscale/swscale.h&gt;</span>
<span class="hl kwb">+#include &lt;libavutil/pixfmt.h&gt;</span>
<span class="hl kwb">+#include &lt;libavutil/fifo.h&gt;</span>
 #define swscale_isRGB(x) ((x)==PIX_FMT_BGR32 || (x)==PIX_FMT_RGB24 \
                         || (x)==PIX_FMT_RGB565 || (x)==PIX_FMT_RGB555 \
                         || (x)==PIX_FMT_RGB8 || (x)==PIX_FMT_RGB4 \
<span class="hl kwd">&#64;&#64; -96,9 +96,10 &#64;&#64;</span>
                         || (x)==PIX_FMT_YUVJ420P || (x)==PIX_FMT_YUVJ422P \
                         || (x)==PIX_FMT_YUVJ444P)
 
<span class="hl kwa">-#define PIX_FMT_ARGB32 PIX_FMT_RGBA32  /* this is just my personal</span>
<span class="hl kwa">-                                        * convenience */</span>
<span class="hl kwb">+#define PIX_FMT_ARGB32 PIX_FMT_RGB32   /* this is just my personal</span>
<span class="hl kwb">+                                         * convenience */</span>
 
<span class="hl kwb">+</span>
 /*
  * file globals
  */
<span class="hl kwd">&#64;&#64; -1248,7 +1249,11 &#64;&#64;</span>
     st-&gt;codec-&gt;flags |= CODEC_FLAG2_FAST;
     // there is no trellis quantiser in libav* for mjpeg
     if (st-&gt;codec-&gt;codec_id != CODEC_ID_MJPEG)
<span class="hl kwb">+#if LIBAVCODEC_VERSION_INT &lt; ((52&lt;&lt;16)+(0&lt;&lt;8)+0)</span>
         st-&gt;codec-&gt;flags |= CODEC_FLAG_TRELLIS_QUANT;
<span class="hl kwb">+#else</span>
<span class="hl kwb">+	st-&gt;codec-&gt;trellis = 1;</span>
<span class="hl kwb">+#endif</span>
     st-&gt;codec-&gt;flags &amp;= ~CODEC_FLAG_OBMC;
     // some formats want stream headers to be seperate
     if (oc-&gt;oformat-&gt;flags &amp; AVFMT_GLOBALHEADER)
<span class="hl kwd">&#64;&#64; -1498,11 +1503,10 &#64;&#64;</span>
         // add the video stream and initialize the codecs
         //
         // prepare stream
<span class="hl kwa">-        out_st =</span>
<span class="hl kwa">-            add_video_stream (output_file, image,</span>
<span class="hl kwa">-                              (input_pixfmt ==</span>
<span class="hl kwa">-                               PIX_FMT_PAL8 ? PIX_FMT_RGB24 : input_pixfmt),</span>
<span class="hl kwa">-                              xvc_codecs[job-&gt;targetCodec].ffmpeg_id, job);</span>
<span class="hl kwb">+        fprintf(stderr, &quot;The current pixfmt is %d, but the choosen one is %d\n&quot;, input_pixfmt, (input_pixfmt == PIX_FMT_PAL8 ? PIX_FMT_RGB24 : input_pixfmt));</span>
<span class="hl kwb">+        out_st = add_video_stream (output_file, image,</span>
<span class="hl kwb">+                     (input_pixfmt == PIX_FMT_PAL8 ? PIX_FMT_RGB24 : input_pixfmt),</span>
<span class="hl kwb">+                      xvc_codecs[job-&gt;targetCodec].ffmpeg_id, job);</span>
 
         // FIXME: set params
         // memset (p_fParams, 0, sizeof(*p_fParams));
Index: src/main.c
===================================================================
<span class="hl kwa">--- src/main.c	(revision 319)</span>
<span class="hl kwb">+++ src/main.c	(working copy)</span>
<span class="hl kwd">&#64;&#64; -45,7 +45,7 &#64;&#64;</span>
 #include &lt;locale.h&gt;
 
 #ifdef USE_FFMPEG
<span class="hl kwa">-#include &lt;ffmpeg/avcodec.h&gt;</span>
<span class="hl kwb">+#include &lt;libavcodec/avcodec.h&gt;</span>
 #endif     // USE_FFMPEG
 
 #include &quot;control.h&quot;
<span class="hl kwd">&#64;&#64; -217,9 +217,6 &#64;&#64;</span>
 
     if (app)
         xvc_appdata_free (app);
<span class="hl kwa">-#ifdef USE_FFMPEG</span>
<span class="hl kwa">-    av_free_static ();</span>
<span class="hl kwa">-#endif</span>
 }
 
 /**

 	  	 
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div id="footer">
	<p>
		Copyright &copy; 2004-2020 aurweb Development Team &ndash;
		<strong>AUR packages are user produced content. Any use of the provided files is at your own risk.</strong>
	</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
